lib/rust/ensogl/core/src/display/object/instance.rs

```diff
@@ -1020,139 +1489,1173 @@ impl InstanceDef {
+    /// the size property if it is set to pixels or to percent of the parent size. Otherwise, set
+    /// the size to zero.
+    fn reset_size_to_static_values<Dim>(&self, x: Dim, parent_size: f32)
+    where Dim: ResolutionDim {
+        let size = match self.layout.size.get_dim(x) {
+            Size::Fixed(unit) => unit.resolve_const_and_percent(parent_size).unwrap_or(0.0),
+            Size::Hug => 0.0,
+        };
+        self.layout.computed_size.set_dim(x, size);
+    }
+
+    fn refresh_layout_internal(&self, pass: LayoutResolutionPass, pass_cfg: PassConfig) {
+        if self.dirty.transformation.check() || self.dirty.modified_children.check_all() {
+            if let Some(layout) = &*self.layout.auto_layout.borrow() && layout.enabled {
+                match pass {
+                    LayoutResolutionPass::First => self.refresh_grid_layout(X, &layout, pass),
+                    LayoutResolutionPass::Second => self.refresh_grid_layout(Y, &layout, pass),
+                }
+            } else {
+                match pass {
+                    LayoutResolutionPass::First => self.refresh_manual_layout(X, pass, pass_cfg),
+                    LayoutResolutionPass::Second => self.refresh_manual_layout(Y, pass, pass_cfg),
+                }
+            }
+        }
+    }
+
+    fn refresh_manual_layout<Dim>(&self, x: Dim, pass: LayoutResolutionPass, pass_cfg: PassConfig)
+    where Dim: ResolutionDim {
+        let hug_children = pass_cfg != PassConfig::DoNotHugDirectChildren;
+        let hug_children = hug_children && self.layout.size.get_dim(x).is_hug();
+        let children = self.children();
+
+        let mut min_x = if children.is_empty() { 0.0 } else { f32::MAX };
+        let mut max_x = if children.is_empty() { 0.0 } else { f32::MIN };
+        let mut children_to_grow = vec![];
+        for child in &children {
+            if child.layout.grow_factor.get_dim(x) > 0.0 {
+                children_to_grow.push(child);
+            } else {
+                child.reset_size_to_static_values(x, self.layout.computed_size.get_dim(x));
+                child.refresh_layout_internal(pass, PassConfig::Default);
+                let child_pos = child.position().get_dim(x);
+                let child_size = child.computed_size().get_dim(x);
+                let child_content_origin = child.content_origin().get_dim(x);
+                let child_min_x = child_pos + child_content_origin;
+                let child_max_x = child_min_x + child_size;
+                min_x = min_x.min(child_min_x);
+                max_x = max_x.max(child_max_x);
+            }
+        }
+        if hug_children {
+            self.layout.computed_size.set_dim(x, max_x - min_x);
+        }
+        for child in children_to_grow {
+            child.layout.computed_size.set_dim(x, self.layout.computed_size.get_dim(x));
+            child.refresh_layout_internal(pass, PassConfig::DoNotHugDirectChildren);
```

Comment:
Is there information somewhere in docs that the grow factor override the "hug" size? This is what I understand from this code.

Reply:
Added explicit info aobut it to docs